SwiftCompilerSources/Sources/Optimizer/Utilities/OptUtils.swift

```diff
@@ -397,6 +397,27 @@ extension LoadInst {
   }
 }
 
+extension PartialApplyInst {
+  var isPartialApplyOfReabstractionThunk: Bool {
+    // A partial_apply of a reabstraction thunk either has a single capture
+    // (a function) or two captures (function and dynamic Self type).
+    if self.numArguments == 1 || self.numArguments == 2 {  
```

Comment:
Again, you can combine those multiple if-statements into one

Reply:
Changed